cni/pkg/iptables/iptables.go

```diff
@@ -141,14 +141,14 @@ func (cfg *IptablesConfigurator) executeDeleteCommands() error {
 	deleteCmds := [][]string{
 		{"-t", iptablesconstants.MANGLE, "-D", iptablesconstants.PREROUTING, "-j", ChainInpodPrerouting},
 		{"-t", iptablesconstants.MANGLE, "-D", iptablesconstants.OUTPUT, "-j", ChainInpodOutput},
-		{"-t", iptablesconstants.NAT, "-D", iptablesconstants.PREROUTING, "-j", ChainInpodPrerouting},
```

Comment:
Is this rule conditionally applied? https://github.com/istio/istio/blob/master/cni/pkg/iptables/iptables.go#L256-L259 The following rules are conditional on `redirectDNS` being enabled

Reply:
You are correct, that rule is no longer optional on master
It was optional last week before tproxy was removed.

I will add it back to the deleteCmds list for 1.25 but then I will do a manual cherry-pick to have it optional in 1.24.x where tproxy is still supported.

Reply:
Note that we may delete rules from 1.24 in a 1.25 binary, and even the inverse! So I support making them all optional

Reply:
Oh right, that could happen.
I will revert and put back the tproxy rule in optional for the time being.
Then I will followup in another commit for 1.25 to make all of them optionals